Reference/Searching/Examine/Quick-Start/index.md

```diff
@@ -66,208 +120,226 @@ Umbraco ships with three indexes:
+The service `IExamineManager` is used to retrieve an Examine index by its 'alias', so we need to inject that service into our `SearchService`.
+```csharp
+using System.Collections.Generic;
+using Examine;
+using Umbraco.Cms.Core.Models.PublishedContent;
 
-```
-@using Examine
-@inject IExamineManager ExamineManager;
-```
- and then get a reference: 
+namespace MyStarterKitSite.Services;
 
-```csharp
- if(ExamineManager.TryGetIndex("ExternalIndex", out var index))
+public class SearchServices : ISearchService
 {
-    var searcher = index.Searcher;
+    private readonly IExamineManager _examineManager;
+    public SearchServices(IExamineManager examineManager)
+    {
+        _examineManager = examineManager;
+    }
+    public IEnumerable<IPublishedContent> SearchContentNames(string query) => throw new NotImplementedException();
+}
 ```
+
 ### Creating the Search Query
-With this in mind we begin to update the `else` condition with the following:
+With the `IExamineManager` injected in our `SearchService`, we can implement the `SearchContentNames` method. We do this using the `Searcher` for the Examine index 'ExternalIndex'.
 
 ```csharp
-else
+IEnumerable<string> ids = Array.Empty<string>();
+if (!string.IsNullOrEmpty(query) && _examineManager.TryGetIndex("ExternalIndex", out IIndex? index))
 {
-    //perform the search
-    //first we try to get the index, it is the ExternalIndex as we don't want to return unpublished things
-    //it returns the index in the var index
-    //be sure to add "@using Umbraco.Examine;" at the top of the view
-    if(ExamineManager.TryGetIndex("ExternalIndex", out var index))
-    {
-        var searcher = index.Searcher;
-        var results = searcher.CreateQuery("content").NodeTypeAlias("person").And().Field("nodeName", searchTerm).Execute();
-        if (results.Any())
-        {
-            <ul>
-                @foreach (var result in results)
-                {
-                    if (result.Id != null)
-                    {
-                        var node = Umbraco.Content(result.Id);
-                        <li>
-                            <a href="@node.Url()">@node.Name</a>
-                        </li>
-                    }
-                }
-            </ul>
-        }
-        else
-        {
-            <p>No results found for query @searchTerm</p>
-        }
-    }
-    return;
+    ids = index
+        .Searcher
+        .CreateQuery("content")
+        .NodeTypeAlias("person")
+        .And()
+        .Field("nodeName", query)
+        .Execute()
+        .Select(x => x.Id);
 }
 ```
-At this point we have chosen to use the External index and it's searcher. 
 
 :::tip
 We reference the External index by it's alias "ExternalIndex". Umbraco has a set of 'Constants' that refer to the indexes that can be more convenient to use `Constants.UmbracoIndexes`. So, in the example here we could have used `Constants.UmbracoIndexes.ExternalIndexName` instead of "ExternalIndex".
```

Comment:
```suggestion
We reference the External index by its alias "ExternalIndex". Umbraco has a set of 'Constants' that refer to the indexes that can be more convenient to use `Constants.UmbracoIndexes`. So, in the example here we could have used `Constants.UmbracoIndexes.ExternalIndexName` instead of "ExternalIndex".
```